src/commands/stac-validate/stac.validate.ts

```diff
@@ -213,6 +207,105 @@ export const commandStacValidate = command({
   },
 });
 
+/**
+ * Validate STAC Assets
+ *
+ * @param stacJson
+ * @param path absolute path of the STAC location
+ * @returns the list of link paths that failed the validation
+ */
+export async function validateAssets(
+  stacJson: st.StacItem | st.StacCollection | st.StacCatalog,
+  path: string,
+): Promise<string[]> {
+  const assetsFailures: string[] = [];
+  const assets = Object.values(stacJson.assets ?? {}) as st.StacAsset[];
+  for (const asset of assets) {
+    const isChecksumValid = await validateStacChecksum(asset, path, { allowMissing: false, allowUnknown: false });
+    if (!isChecksumValid) {
+      assetsFailures.push(path);
+    }
+  }
+  return assetsFailures;
+}
+
+/**
+ * Validate STAC Links
+ *
+ * @param stacJson
+ * @param path absolute path to the STAC location
+ * @returns the list of link paths that failed the validation
+ */
+export async function validateLinks(
+  stacJson: st.StacItem | st.StacCollection | st.StacCatalog,
+  path: string,
+): Promise<string[]> {
+  const linksFailures: string[] = [];
+  for (const link of stacJson.links) {
+    if (link.rel === 'self') continue;
+    // we `allowMissing` because some STAC links might not have a checksum yet (feature added later)
```

Comment:
I don't understand this comment. Why would any STAC links not have a checksum at this point? Can we enumerate them, so that we can avoid this flag? Also, what's meant by "feature added later"?

Reply:
The might not have checksums if the STAC has been generated before the feature that add a checksum for the links. As far as I know, we don't have a process to regenerate the STAC only from an existing dataset.